### ISSUE TITLE
Caster GUI settings window does not launch

### ISSUE THREAD

**Issue by LexiconCode** [synkarius/caster#327]
Caster has a GUI for settings does not launch. To test use use use Spec "launch settings window"
```
ERROR:action.exec:Execution failed: Timeout while waiting for window context: WaitWindow(title='caster settings window v 0.5.11')
ERROR:action.exec:Execution failed: Failed to find window (title='Caster Settings Window v 0.5.11').
```

**Comment by LexiconCode** [comment]
Works prior to  Add default format and secondary format (#262) `

**Comment by mrob95** [comment]
I'm getting this on my normal branch, which is way behind #262. Increasing the time out didn't fix it.

**Comment by LexiconCode** [comment]
Remember to delete the settings.toml

**Comment by mrob95** [comment]
Still getting the same thing in both the `master` and `develop` branches

**Comment by LexiconCode** [comment]
![capture](https://user-images.githubusercontent.com/24551569/48860656-201dec80-ed87-11e8-974c-9d1a974976eb.JPG)
1.   It works after I deleted Caster and re-cloned. 
       Reverted to 7ddf6409e30d76b5f5736493efbbdf95a582dbb1
       deleted .toml files in data directory
- UI Works as expected
2. Reset branch to commit #262 
- UI Fails with the error above.
3. Reverted to 7ddf6409e30d76b5f5736493efbbdf95a582dbb1 commit without deleting .toml from data
- UI Fails with the error above.
4. Still using 7ddf6409e30d76b5f5736493efbbdf95a582dbb1 but deleted all time all .toml files
- UI Works as expected

Therefore I'm left with the conclusion that UI cannot parse some of the new fields.

**Comment by LexiconCode** [comment]
I cant imagine what would be causing it to be different on both of our machines.

**Comment by mrob95** [comment]
This is what i'm seeing on the master branch. Manually modifying the settings files has fixed it on my main branch though... I am confused.

![settings error](https://user-images.githubusercontent.com/42875462/48862369-1794d880-edbe-11e8-96d2-a60a8ea01880.jpg)

**Comment by LexiconCode** [comment]
Okay I'm not sure what to say but it works on the main branch for me as well. Hmmm it's going requires some thought. In addition a few other people to test would be helpful to see a trend.

**Comment by comodoro** [comment]
I can confirm. From inspecting the code it looks like the settings window traverses the settings file for all settings and lists them in the GUI, assuming that all lists contain strings only. I did not know that, all the more reason to move the format definitions outside the settings file where they frankly don't belong. I was thinking about may be introducing a default TOML file for CCR grammars for this purpose( and may be for listing default enabled grammars, like the core ones )

**Comment by LexiconCode** [comment]
Yeah it's one of those undocumented features I forgot about over the mists of time. So I think it's a good plan to move it out into its own file. There's really no rush.

**Comment by Versatilus** [comment]
If the goal is to make the settings files human editable then I think it would make more sense to have the format lists as descriptive strings rather than  plain integers.
```
[formats.HTML]
secondary_format = [ "laws", "spine",]
text_format = [ "laws", "none",]
```
instead of
```
[formats.HTML]
secondary_format = [ 5, 2,]
text_format = [ 5, 0,]
```

**Comment by LexiconCode** [comment]
That does indeed make a lot of sense.

**Comment by comodoro** [comment]
Indeed. I have just broken down the lists, like

```
[formats.Java.secondary_format]
spacing = 1
capitalization = 2
```

which is a little more verbose, but avoids lists entirely. I might use the strings here?

**Comment by comodoro** [comment]
For clarification, I mean the strings with this setup:

```
[formats.Javascript.text_format]
spacing = "gum"
capitalization = "Gerrish"

[formats.Javascript.secondary_format]
spacing = "gum"
capitalization = "tie"
```

Question: "atom_palette_wait" being string is just some kind of typo right? A detail, but might confuse something.

**Comment by LexiconCode** [comment]
"atom_palette_wait" was not intended as a typo. although I agree it should be consistent as an integer if it works as intended when converted over. Try converting it to an integer in the settings and convert it in caster\apps\atom.py as well. 
@comodoro
